### PR TITLE
plugins: npm

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

Port the `npm` plugin from [`snapcraft`](https://github.com/snapcore/snapcraft/blob/main/snapcraft/plugins/v2/npm.py) to `craft-parts`.

CRAFT-957

### CODE REVIEW

**Review by sergiusens — CHANGES_REQUESTED** [review]
I have been thinking about this one some more, and I think we might just be better off bringing in the plugin verbatim as this will produce a resulting part that cannot run as node is never really copied into CRAFT_PART_INSTALL. If you have a better idea I am all ears :-)

**Comment by mr-cal** [comment]
As discussed, I added a new flag called `npm-include-node`, which defaults to False.
Setting it to true will download and include `node`.

File: craft_parts/plugins/npm_plugin.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        """Return a set of required snaps to install in the build environment."""
+        return set()
+
+    def get_build_packages(self) -> Set[str]:
+        """Return a set of required packages to install in the build environment."""
+        return {"curl", "gcc"}
+
+    def get_build_environment(self) -> Dict[str, str]:
+        """Return a dictionary with the environment to use in the build step."""
+        return dict(PATH="${CRAFT_PART_INSTALL}/bin:${PATH}")
```
Comment (sergiusens) [inline craft_parts/plugins/npm_plugin.py:151]: this should only be set if options.npm_include_node is True, right?

Comment (mr-cal) [reply craft_parts/plugins/npm_plugin.py]: Right. And the same is true for `curl` for `get_build_packages()`.  
I added logic and unit tests for both.

**Review by sergiusens — APPROVED** [review]

**Pull request merged**